CRM/Core/Permission/Joomla.php

```diff
@@ -131,4 +135,74 @@ public function checkGroupRole($array) {
     return FALSE;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function upgradePermissions($permissions) {
```

Comment:
I'm not sure how `@inheritDoc` works -- not sure it's doing anything in my IDE -- but the `$permissions` array is documented in the parent class (in the overridden method) as 'Same format as CRM_Core_Permission::getCorePermissions().' I can add a comment to my code if that's not sufficient.